sphinx4-core/src/main/java/edu/cmu/sphinx/fst/Fst.java

```diff
@@ -401,13 +401,26 @@ public boolean equals(Object obj) {
     /*
      * (non-Javadoc)
      * 
+     * @see java.lang.Object#hashCode()
+     */
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(isyms) + Arrays.hashCode(osyms) +
```

Comment:
Previously it was advised to use polynomial hash algorithm as the default one, but my knowledge may be outdated here.